apollo-test/src/main/java/com/spotify/apollo/test/StubClient.java

```diff
@@ -275,13 +319,48 @@ private ResponseSource responseSource() {
 
   }
 
+
+  public class WhenResponseBuilder {
```

Comment:
As objects of this class should not be reused, I think supplying both arguments to addMatcherFirst at the same time is a little bit clearer.
If this class will remain I think it could have a different name as 'builder' may be misleading considering that it modifies the StubClient.